src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -282,21 +290,59 @@ public DeconstructionVariable(ArrayBuilder<DeconstructionVariable> variables, CS
         }
 
         /// <summary>
-        /// For cases where the RHS of a deconstruction-assignment has no type (TupleLiteral), we squint and look at the LHS as a tuple type to give the RHS a type.
+        /// For cases where the RHS of a deconstruction-declaration is a tuple literal, we merge type information from both the LHS and RHS.
+        /// For cases where the RHS of a deconstruction-assignment is a tuple literal, the type information from the LHS determines the merged type, since all variables have a type.
```

Comment:
We never have to check explicitly. The LHS in an assignment is always completely typed (no vars).